functions.js

```diff
@@ -0,0 +1,106 @@
+        } else {
+            // Unique Stations on Dropdown requires selection of all instances of Origins and Destinations...
+            const origins = await getStationInstances(originSelection)
+            const destinations = await getStationInstances(destinationSelection)
+
+            // Extract Lines to See if Single Line is Possible
+            const originLines = origins.map(station => station.line)
+            const destinationLines = destinations.map(station => station.line)
+
+            // Create Pairs of Origins and Destinations. Stations Exist across multiple Lines.
+            const pairs = originLines.reduce((accumulator, line, originIndex) => {
+                const destinationIndex = destinationLines.indexOf(line)
+                if (destinationIndex !== -1) {
+                    accumulator.push([originIndex, destinationIndex])
+                }
+                return accumulator
+            }, [])
+
+            // Create Routes populated with Origin and Destination station Details
+            let routes = pairs.map(([originIndex, destinationIndex]) => {
+                const originStation = origins[originIndex];
+                const destinationStation = destinations[destinationIndex];
+
+                return {
+                    line: originStation.line,
+                    start: Math.min(originStation.position, destinationStation.position),
+                    end: Math.max(originStation.position, destinationStation.position)
+                }
+            })
+
+            //Query Journey Data
+            const journeys = []
+            for (const route of routes) {
+                let journey = await getSingleLineJourney(route.line, route.start, route.end);
+                journeys.push(journey);
+            }
+
+            //Summarise data to provide information in the format of end Route Card
+            const journeySummaries = []
+
+            for (const journey of journeys) {
+
+                let journeyArrayLength = journey.length - 1
+
+                if (journey[0].code === originSelection) {
+                    const from = journey[0].name
+                    const to = journey[journeyArrayLength].name
+                    const line = journey[0].line
+                    const stationBreakdown = journey.map(station => station.name)
+                    journeySummaries.push({
+                        from: from, to: to, line: line, stations: stationBreakdown
+                    })
+                } else {
+                    const from = journey[journeyArrayLength].name
+                    const to = journey[0].name
+                    const line = journey[journeyArrayLength].line
+                    const stationBreakdown = journey.map(station => station.name).reverse()
+                    journeySummaries.push({
+                        from: from, to: to, line: line, stations: stationBreakdown
+                    })
```

Comment:
There's a fair bit of duplication here, so at least some of this could be split out into a function